### PR TITLE
gtk4: update to 4.16.2

### PR DESCRIPTION
**Pull request by chrysos349**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86-64)

`mesa-24.2.2` (#51881) needs to be merged first. see the note below.
```
Note: This release changes the default GSK renderer to be Vulkan,
on Wayland. Other platforms still use ngl. The intent of this change
is to use the best available platform APIs. You can still override
the renderer choice using the GSK_RENDERER environment variable.

We believe that most of the problems reported with the new renderers
during the 4.13 and 4.15 development cycles have been addressed by now.

But the new renderers and dmabuf support are using graphics drivers
in different ways than the old gl renderer, and trigger new driver bugs.
Therefore, it is recommended to use the latest mesa release (24.2)
with the new renderers.
```

### CODE REVIEW

**Comment by oreo639** [comment]
Remove the extraneous tracker commit.

**Comment by chrysos349** [comment]
> Remove the extraneous tracker commit.

it won't compile otherwise. see the error :

```
meson.build:445:15: ERROR: Dependency lookup for tracker-sparql-3.0 with method 'pkgconfig' failed: Could not generate cflags for tracker-sparql-3.0:
Package sqlite3 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sqlite3.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sqlite3', required by 'tracker-sparql-3.0', not found
```

**Comment by oreo639** [comment]
Ah, my bad. Thanks.

File: srcpkgs/gtk4/template [hunk 1]
```diff
@@ -31,9 +31,9 @@ maintainer="Orphaned <orphan@voidlinux.org>"
 license="LGPL-2.1-or-later"
 homepage="https://www.gtk.org/"
 #changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/main/NEWS"
-changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/gtk-4-14/NEWS"
+changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/4.16.0/NEWS"
```
Comment (oreo639) [inline srcpkgs/gtk4/template:34]: This should be ` changelog="https://gitlab.gnome.org/GNOME/gtk/-/raw/gtk-4-16/NEWS"`

**Pull request merged**